dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/cookies/SameSiteCookieUtils.java

```diff
@@ -0,0 +1,277 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.http.channel.internal.cookies;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.ffdc.FFDCFilter;
+import com.ibm.ws.http.channel.internal.HttpMessages;
+
+/**
+ *
+ */
+public class SameSiteCookieUtils {
+
+    /** RAS tracing variable */
+    private static final TraceComponent tc = Tr.register(CookieUtils.class,
+                                                         HttpMessages.HTTP_TRACE_NAME,
+                                                         HttpMessages.HTTP_BUNDLE);
+
+    private static final int UNSUPPORTED_IOS_VERSION = 12;
+    private static final int UNSUPPORTED_MACOS_MAJOR_VERSION = 10;
+    private static final int UNSUPPORTED_MACOS_MINOR_VERSION = 14;
+    private static final Pattern IosVersion = Pattern.compile(".*\\(ip.+; cpu .*os (\\d+)[_\\d]*.*\\) applewebkit\\/.*");
+    private static final Pattern MacOsVersion = Pattern.compile(".*\\(macintosh;.*mac os x (\\d+)_(\\d+)[_\\d]*.*\\) applewebkit\\/.*");
+    private static final Pattern Safari = Pattern.compile(".*version\\/.* safari\\/.*");
+    private static final Pattern MacEmbeddedBrowser = Pattern.compile("^mozilla\\/[\\.\\d]+ \\(macintosh;.*mac os x [_\\d]+\\) applewebkit\\/[\\.\\d]+ \\(khtml, like gecko\\)$");
+
+    private static final Pattern ucBrowser = Pattern.compile(".*ucbrowser/.*");
+    private static final Pattern ucBrowserVersion = Pattern.compile(".*ucbrowser/(\\d+)\\.(\\d+)\\.(\\d+)[\\.\\d]*.*");
+
+    private static final int UC_MAJOR_VERSION = 12;
+    private static final int UC_MINOR_VERSION = 13;
+    private static final int UC_BUILD_VERSION = 2;
+
+    private static final Pattern chromiumBased = Pattern.compile(".*chrom(e|ium).*");
+    private static final Pattern chromiumVersion = Pattern.compile(".*chrom[^\\/]+\\/(\\d+)[\\.\\d]*.*");
+
+    private static final int CHROMIUM_MIN_MAJOR_VERSION = 51;
+    private static final int CHROMIUM_MAX_MAJOR_VERSION = 66;
+
+    /**
+     * Default constructor for the utility class.
+     */
+    private SameSiteCookieUtils() {
+        // nothing to do
+    }
+
+    /**
+     * Compares the request's user agent against a regex of known incompatible
+     * clients for the SameSite=None cookie attribute. If the client is incompatible,
+     * this method will return true.
+     *
+     * @param userAgent
+     * @return
+     */
+    public static boolean isSameSiteNoneIncompatible(String userAgent) {
+        System.out.println("isSameSiteNoneIncompatible: enter");
```

Comment:
I assume we'll move this to an actual logger of some sort and not use System.out?